src/data/reports/glass-data-submission/GLASSDataSubmissionDefaultRepository.ts

```diff
@@ -725,63 +745,116 @@ export class GLASSDataSubmissionDefaultRepository implements GLASSDataSubmission
+
+                _.forEach(amrIndividualQuestionnaires, async amrIndividualQuestionnaire => {
+                    await this.duplicateDataSet(amrIndividualQuestionnaire, items);
+                });
                 _.forEach(amrIndividualPrograms, async amrIndividualProgram => {
-                    await this.duplicateProgramStages(amrIndividualProgram, items);
+                    await this.duplicateTrackerProgram(amrIndividualProgram, items);
                 });
+                break;
             }
-            if (module === "EGASP") {
-                const egaspPrograms = modules.find(module => module.name === "EGASP")?.programs ?? [];
-                _.forEach(egaspPrograms, async egaspProgram => await this.duplicateProgram(egaspProgram, items));
+            case EGASP: {
+                const egaspPrograms = modules.find(module => module.name === EGASP)?.programs ?? [];
+                _.forEach(
+                    egaspPrograms,
+                    async egaspProgram =>
+                        await this.duplicateEventProgram(egaspProgram, items, selectedModule === EGASP)
+                );
+                break;
             }
+        }
+    }
 
-            const newSubmissionValues = this.getNewSubmissionValues(items, objects, "APPROVED");
-            const recipients = await this.getRecipientUsers(items, modules);
-
-            const message = await this.getNotificationText(items, modules, "approved");
-            this.sendNotifications(message, message, [], recipients);
-
-            return await this.globalStorageClient.saveObject<GLASSDataSubmissionItem[]>(namespace, newSubmissionValues);
-        } else if (signals) {
-            const objects = await this.globalStorageClient.listObjectsInCollection<EARDataSubmissionItem>(namespace);
-            const newSubmissionValues = this.getNewEARSubmissionValues(signals, objects, "APPROVED");
-            const recipients = await this.getEARRecipientUsers(signals, modules);
-
-            const message = this.getEARNotificationText(signals, modules, "approved");
-            this.sendNotifications(message, message, [], recipients);
+    private async getAMCProductRegisterId() {
+        const { objects } = await this.api.models.programs
+            .get({
+                fields: { id: true },
+                filter: { name: { eq: "AMC - Product Register" } },
+            })
+            .getData();
 
-            return await this.globalStorageClient.saveObject<EARDataSubmissionItem[]>(namespace, newSubmissionValues);
-        }
+        return _.first(objects)?.id ?? "";
     }
 
     private async duplicateDataSet(dataSet: ApprovalIds, items: GLASSDataSubmissionItemIdentifier[]) {
-        await promiseMap(items, async item => {
+        _.forEach(items, async item => {
             const dataValueSets = (await this.getDataSetsValue(dataSet.id, item.orgUnit ?? "", item.period)).dataValues;
```

Comment:
if the orgUnit is "" , ideally we should not duplicate the dataset right? Better to show an error and exit if orgUnit is "". Is there any valid situation where orgUnit would be "" ?